File: problems/best-time-to-buy-and-sell-stock-ii/solution-3.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  let maxProfit = 0
  for (let i = 0; i < prices.length; i++) {
    if (prices[i] > prices[i - 1]) {
      maxProfit += prices[i] - prices[i-1]
    }
  }
  return maxProfit
}

File: problems/best-time-to-buy-and-sell-stock-ii/solution-3.py
class Solution:
    def maxProfit(self, prices: [int]) -> int:
        maxProfit = 0
        for i in range(1, len(prices)):
            if prices[i] > prices[i-1]:
                maxProfit += (prices[i] - prices[i-1])
        return maxProfit